packages/protocol/contracts/bridge/libs/LibBridgeProcess.sol

```diff
@@ -80,14 +80,20 @@ library LibBridgeProcess {
         // We retrieve the necessary ether from EtherVault if receiving on
         // Taiko, otherwise it is already available in this Bridge.
         address ethVault = resolver.resolve("ether_vault", true);
-        if (ethVault != address(0)) {
+        if (
+            ethVault != address(0) &&
+            (message.depositValue + message.callValue + message.processingFee >
```

Comment:
The addition of the 3 values (`depositValue, callvalue, processingFee`) executed twice. Slightly cheaper if we use a separate, temp value for that and use it.
`uint256 allValue = message.depositValue + message.callValue + message.processingFee;`